Doc/whatsnew/3.12.rst

```diff
@@ -1895,6 +1895,15 @@ Changes in the Python API
   * Mixing tabs and spaces as indentation in the same file is not supported anymore and will
     raise a :exc:`TabError`.
 
+  * The :mod:`threading` module now expects the :mod:`!_thread` module to have
+    a ``_is_main_interpreter`` attribute.  Is is a function with no
+    arguments that returns ``True`` if the current interpreter is the
+    main interpreter.
+
+    Any library or application that provides a custom ``_thread`` module
+    must make sure it provides ``_is_main_interpreter()``.
```

Comment:
```suggestion
    should provide ``_is_main_interpreter()``.
```

Let's weaken the language since it's not actually something we can (or should) require in 3.12.